client/js/heat.js

```diff
@@ -0,0 +1,91 @@
+class HeatMapRow {
+  /**
+   * format for a row for heat map values obj
+   * @param {number[]} data binary arr of days where music was listened to
+   */
+  constructor(data) {
+    this.data = data;
+  }
+}
+
+// TODO: retrieve data from the backend instead of using a hardcoded value
+const TOTAL_LIKED = 13;
+// list of day indices where music was played
+const LIKED_MUSIC_HIST = [1, 4, 7];
+
+// ith value of 1 means ith latest video is music
+const likedMusicBinaryHist = makeBinaryArr(LIKED_MUSIC_HIST, TOTAL_LIKED);
+
+/**
+ * converts an array to binary arr
+ * @param {number[]} arr values will be indices that hold 1 in output arr
+ * @param {number[]} total size of output arr
+ * @returns binary arr of size total where ith value 1
+ * corresponds to value i in input arr
+ */
+function makeBinaryArr(arr, total) {
```

Comment:
This function name is vague and just tells us about our data type. Given that you already use terms like `heatIndex` in your function body, you can make the name more specific (like `constructHeatMapSource`) even though the function itself is more general than this.